packages/manager/modules/product-offers/src/workflows/product-offers-order-workflow.class.js

```diff
@@ -0,0 +1,246 @@
+import get from 'lodash/get';
+import isEmpty from 'lodash/isEmpty';
+import isFunction from 'lodash/isFunction';
+import isString from 'lodash/isString';
+
+import Workflow from './product-offers-workflow.class';
+
+import { CATALOG_ITEM_TYPE_NAMES } from './product-offers-workflow.constants';
+
+/**
+ * Workflow Class to handle option order
+ */
+export default class OrderWorkflow extends Workflow {
+  /**
+   * @param {Object} $translate      AngularJS provider
+   * @param {Object} workflowOptions Specific options for this workflow,
+   *  must contains the following values:
+   *  - {catalog}: Catalog to use from GET '/order/catalog/public/{productName}',
+   *  - {catalogItemTypeName}: Item type of which we want catalog information,
+   *   as the catalog contains 'addons' and 'plans' where product information
+   *  can be found.
+   *  - {productName}: The product name to use, to add items for item order.
+   *  e.g.: 'webHosting',
+   *  - {serviceNameToAddProduct}: Service name of which we will add product/addon.
+   *  If set, the order will consist to add option to an existing service.
+   *  If null, the order concerns an new product.
+   *  - {onBeforePricingGetPlancode}: Method called to get the product/addon plan
+   *  code, which will be called after configuration phase, and before pricing
+   *  step. Plan code example : webHosting, cloudDB.
+   *  - {onGetConfiguration}:
+   *  Method to get configuration items that will be added to the order
+   *  cart, called when fetching checkout information.
+   *  The configuration key list must follow catalog product required
+   *  configuration.
+   *  This can be found through, GET /order/catalog/public/{productName},
+   *  in plans > planItem > configurations.
+   *  Example for cloudDB (GET /order/catalog/public/cloudDB, with FR
+   *  subsidiary):
+   *  configurations = [
+   *    {
+   *      isCustom: false,
+   *      values: [ "gra1" ],
+   *      name: "dc",
+   *      isMandatory: true
+   *    },
+   *    {
+   *      name: "engine"
+   *      values: [
+   *        "redis_3.2"
+   *        "redis_4.0"
+   *        ...
+   *      ],
+   *      isCustom: false,
+   *      isMandatory: true
+   *    }
+   *  ]
+   *  Here, the configuration label will be the 'name' property value,
+   *  and the configuration value, one of the 'values' list.
+   *  Returned value must be of the following format:
+   *  [
+   *    {
+   *      label: configurationLabel (ex: 'legacy_domain'),
+   *      value: configurationValue (ex: 'www.ovhcloud.com'),
+   *    }
+   *  ];
+   * @param {Object} WucOrderCartService Service to handle order cart
+   */
+  /* @ngInject */
+  constructor($q, $translate, workflowOptions, WucOrderCartService) {
+    super($q, $translate, workflowOptions);
+    this.WucOrderCartService = WucOrderCartService;
+
+    if (!this.catalog) {
```

Comment:
I'm having trouble knowing where `this.catalog` comes from

Reply:
Welcome to the world of OOP

Reply:
@marie-j built by base class constructor